src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5033,38 +5032,39 @@ void SessionImpl::loadCategories()
 {
     m_categories.clear();
 
-    QFile confFile {(specialFolderLocation(SpecialFolder::Config) / CATEGORIES_FILE_NAME).data()};
-    if (!confFile.exists())
-    {
-        // TODO: Remove the following upgrade code in v4.5
-        // == BEGIN UPGRADE CODE ==
-        upgradeCategories();
-        m_needUpgradeDownloadPath = true;
-        // == END UPGRADE CODE ==
-
-//        return;
-    }
+    const int fileMaxSize = 1024 * 1024;
+    const Path path = specialFolderLocation(SpecialFolder::Config) / CATEGORIES_FILE_NAME;
 
-    if (!confFile.open(QFile::ReadOnly))
+    const auto readResult = Utils::IO::readFile(path, fileMaxSize);
+    if (!readResult)
     {
-        LogMsg(tr("Failed to load Categories. File: \"%1\". Error: \"%2\"")
-               .arg(confFile.fileName(), confFile.errorString()), Log::CRITICAL);
+        if (readResult.error().status == Utils::IO::ReadError::ErrorStatus::NotExist)
```

Comment:
So I would change it to be `enum` instead of `enum class`. Or at least (also taking into account https://github.com/qbittorrent/qBittorrent/pull/19095#discussion_r1224683914) rename it to something like `Code` (aka `ReadError::Code`).